slog/slog.go

```diff
@@ -598,3 +602,14 @@ func Emergencyf(format string, v ...interface{}) {
 func (e *Entry) Emergencyf(format string, v ...interface{}) {
 	e.logger.log(e, severityEmergency, fmt.Sprintf(format, v...), 2)
 }
+
+// NewContext returns a new Context that carries an entry.
+func NewContext(ctx context.Context, entry *Entry) context.Context {
+	return context.WithValue(ctx, entryKey, entry)
+}
+
+// FromContext returns the Entry value stored in ctx, if any.
+func FromContext(ctx context.Context) (*Entry, bool) {
+	entry, ok := ctx.Value(entryKey).(*Entry)
```

Comment:
While the docs use return of '(*User, bool)' as an example, I think we should go with just '*Entry', defaulting to returning a new entry if one is not available. This will allow calling functions to know that they can always just use the entry returned and not have to worry about checking validity.